Tests/Unit/DependencyInjection/HttplugExtensionTest.php

```diff
@@ -267,4 +268,145 @@ private function verifyProfilingDisabled()
             );
         }
     }
+
+    public function testClientShouldHaveDefaultVisibility()
+    {
+        $this->load([
+            'clients' => [
+                'acme' => [],
+            ],
+        ]);
+
+        $this->assertContainerBuilderHasService('httplug.client.acme');
+
+        if (version_compare(Kernel::VERSION, '3.4', '>=')) {
```

Comment:
The version of the HttpKernel component does not necessarily have to be the same as the DependencyInjection component. If we want to fully be on the safe side, we could for example check for the `EnvVarProcessor` class to exist:

```php
use Symfony\Component\DependencyInjection\EnvVarProcessor;

// ...

if (class_exists(EnvVarProcessor::class)) {
    // ...
}
```